executable-spec/src/Cardano/Ledger/Spec/STS/Update/Data.hs

```diff
@@ -304,35 +306,29 @@ instance HasTypeReps URL where
     $ typeOf (undefined :: URL)
       : replicate (T.length text) (typeOf (undefined :: Char))
 
-deriving instance ( Typeable hashAlgo
-                  , HasTypeReps (SIP hashAlgo)
-                  , HasTypeReps hashAlgo
-                  , HasTypeReps (Hash hashAlgo SIPData)
-                  ) => HasTypeReps (IdeationPayload hashAlgo)
+type DataHasTypeReps crypto = (Crypto crypto, HasTypeReps (Hash (HASH crypto) SIPData))
```

Comment:
Yeah, naming here seems a bit off. I'd maybe go with something like `CryptoWithDataTypeReps`, but maybe that's just too long winded.

Reply:
I like that one better. Thanks!